### ISSUE TITLE
The implementation of GMAE loss

### ISSUE THREAD

**Issue by rowedenny** [Teng-Sun/GEAR#2]
Hi,

Thank you for this amazing work extending GCE to more applications.

I have a question for the implementation of GMAE loss of Eq7. Per the implementation, I find the following snippet
```
class GeneralizedCELoss(nn.Module):

    def __init__(self, args, q):
        super(GeneralizedCELoss, self).__init__()
        self.args = args
        self.q = q
        self.criterion = nn.L1Loss(reduction='none')
             
    def forward(self, logits, targets):
        loss_l1 = self.criterion(logits.detach(), targets)
        if self.args.gce == 'tanh':
            loss = torch.tanh((1/loss_l1) ** self.q) 
        elif self.args.gce == 'sigmoid':
            loss = 2 * torch.sigmoid(-loss_l1)
        elif self.args.gce == 'test1':
            loss =  torch.exp(-loss_l1 / self.q)
        elif self.args.gce == 'test2':
            loss = 1 / (1 + self.q * self.q * loss_l1 * loss_l1) 
        else:
            loss = 0
        loss = loss * self.criterion(logits, targets)
        return loss 
```

If I understand correctly, this implementation re-weight the gradient of the naive criterion, where `loss` at the end of the implementation is the re-weigh weighting, corresponding to the coeff in Eq8. Is that correct?
Thank you.

**Comment by Teng-Sun** [comment]
Thank you for your attention to our work. We referred to Eq. 8 during the code implementation, and self.args.gce should be 'sigmoid'.

**Comment by rowedenny** [comment]
Thanks for the response, which has addressed my question.
